### ISSUE TITLE
Auto Position Functionality

### ISSUE THREAD

**Issue by jshcrowthe** [jshcrowthe/dialog-el#14]
It has been expressed that auto position functionality _(see: #13)_ would be useful for this element.

For the API of this element I intend to follow the native `dialog` element pattern on the `show` function _(see: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog)_

So it'd look something like the following:

```html
<dialog-el>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipisicing elit. 
    Velit vero, ea! Magni voluptatem dolorum, hic omnis saepe. 
    Ad veritatis, suscipit quia distinctio fuga, ab, velit 
    tempore nobis modi consequuntur nostrum.
  </p>
</dialog-el>

<button>Open Me!</button>
```

```js
var dialog = document.querySelector('dialog-el');
var button = document.querySelector('button');

dialog.show(button);
```

The dialog would then fly out to the space of the screen w/ the most space relative to the passed element.

**Comment by Norman-Amy** [comment]
I mentioned this in the PR, but just to reiterate it here, when we pass in the element, it will have offsetTop/Left/Height/Width properties that will make it possible for us to compute the positioning of the dialog. We will also need to keep the width of the dialog in mind to ensure that it will fit on the screen.

Another thing worth noting is that the `arrow-direction` on the dialog should default to what is specified by the container attribute and it will only be changed when the dialog doesn't fit in the space available to it. In that case, `arrow-direction="left"` should become `arrow-direction="right"`  and `arrow-direction="top"` should become `arrow-direction="bottom"` and vice versa. I guess in the use case where a screen is too small in any direction (like on mobile), we will need to figure something else out.

**Comment by Norman-Amy** [comment]
One problem I've run into when experimenting with this: how to get the target element to the show function. For example, we use an element in a parent component to open the person-card, and then person-card calls the dialog's show function. How do we pass the element from the parent component to the person-card so it can be passed to the dialog?

**Comment by liechtym** [comment]
@jshcrowthe @Norman-Amy Any update on this? I've written another dialog element with a very simple auto-positioning algorithm similar to what you are describing. Perhaps I can help contribute?

**Comment by Norman-Amy** [comment]
@jshcrowthe said he had some feedback a couple weeks ago. Until this or a refactored version of this is in, we are pinned to my fork . Suggestions are always welcome, if you have some.

**Comment by jshcrowthe** [comment]
I have reviewed @Norman-Amy's PR and had some questions on some design issues. Those are documented in #20.

I have also created a PR (can be found here: #23) that needs some review from those interested but is a proposed implementation of auto positioning.

**Comment by liechtym** [comment]
@jshcrowthe I made a PR to your auto-position branch (https://github.com/jshcrowthe/dialog-el/pull/25) with an idea for positioning the dialog relative to the document instead of the window or offsetParent. Let me know what you think of this idea. I would also like your feedback @Norman-Amy

**Comment by jshcrowthe** [comment]
Closed in #23
